gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/process/traversal/strategy/optimization/FilterRankingStrategyTest.java

```diff
@@ -106,7 +106,7 @@ public void doTest() {
                 {__.dedup().has("value", 0).or(not(has("age")), has("age", 10)).has("value", 1), __.has("value", 0).has("value", 1).or(not(has("age")), has("age", 10)).dedup(), Collections.singletonList(InlineFilterStrategy.instance())},
                 {__.dedup().filter(out()).has("value", 0), has("value", 0).filter(out()).dedup(), Collections.emptyList()},
                 {filter(out()).dedup().has("value", 0), has("value", 0).filter(out()).dedup(), Collections.emptyList()},
-                {__.as("a").out().has("age").where(P.eq("a")), __.as("a").out().where(P.eq("a")).has("age"), Collections.emptyList()},
+                {__.as("a").out().has("age").where(P.eq("a")), __.as("a").out().has("age").where(P.eq("a")), Collections.emptyList()},
```

Comment:
`where(eq('a')` might be faster, but probably not. But we don't know what the `P` predicate is. If its something more complex, than `==` then it will be slower. And no, `ReferenceVertex` doesn't come into play here as those are resolved at "attachment" and not dynamically in mid-traversal.